### ISSUE TITLE
impl iterator for ObjectList

### ISSUE THREAD

**Issue by clux** [clux/kube-rs#41]
useability. easy to do. But [requires a bunch of sibling traits](https://www.reddit.com/r/rust/comments/7a0slp/questionimplementing_iterator_for_a_struct_with_a/dp6a77s/) (option 1).

**Comment by kitfre** [comment]
I'd be interested in taking a shot at this! After reading the linked reddit comment I'm still a bit unclear, this is what I have so far -- [is this the direction you were imagining](https://github.com/kitfre/kube-rs/blob/0602c20f9184d7a7baffdb16c28872d5259eb6b4/src/api/resource.rs#L122)? 
From the linked line down is an implementation of `Iterator` using a companion struct, and `IntoIterator` using `std::vec::IntoIter`

**Comment by clux** [comment]
Hey, thanks for this!

I thought this was the best solution initially, but now slowly realising it's so many traits to implement and maintain. I think it's probably better to just do the easy solutions (option 2 or 3).

Think 2 is is just something like:

```rust
impl<T> ObjectList<T>
    pub fn iter<'a>(&'a self) -> impl Iterator<Item = &T> + 'a {	
        self.items.iter()
    }
}
````

For the one hard one ^

..then probably an `impl IntoIterator for ObjectList<T>` and one that allows calling it on a burrowed objectlist. See https://github.com/babylonhealth/shipcat/commit/ce9b0e5c3 (some of the removed code near below that implements iter).

Solution 3 would be just the deref i guess? I would need to see how it would look in the examples. Can we call `for p in pod.list()` or would we have to insert explicit stars to deref? If so, then option 2 is probably nicer.

**Comment by kitfre** [comment]
After reading through the docs for `Deref` I went with Option 2 instead, it felt a bit cleaner. I've also added `iter_mut` and an implementation of `IntoIterator` for a mutable reference.

**Comment by clux** [comment]
Closed by #91
